src/components/results/dynamicsimulation/dynamic-simulation-result-time-line.tsx

```diff
@@ -0,0 +1,212 @@
+        colKey: COL_TIME,
+        sortWay: SORT_WAYS.asc,
+    });
+
+    const { updateFilter, filterSelector } = useAggridLocalRowFilter(gridRef, {
+        [COL_TIME]: COL_TIME,
+        [COL_MODEL_NAME]: COL_MODEL_NAME,
+        [COL_MESSAGE]: COL_MESSAGE,
+    });
+
+    const sortAndFilterProps = useMemo(
+        () => ({
+            sortProps: {
+                onSortChanged,
+                sortConfig,
+            },
+            filterProps: {
+                updateFilter,
+                filterSelector,
+            },
+        }),
+        [onSortChanged, sortConfig, updateFilter, filterSelector]
+    );
+
+    // columns are defined from fields in {@link TimelineEvent} types
+    const columnDefs = useMemo(() => {
+        return [
+            makeAgGridCustomHeaderColumn({
+                headerName: intl.formatMessage({
+                    id: 'DynamicSimulationTimeLineEventTime',
+                }),
+                field: COL_TIME,
+                width: MIN_COLUMN_WIDTH,
+                filterParams: {
+                    filterDataType: FILTER_DATA_TYPES.NUMBER,
+                    filterComparators: Object.values(FILTER_NUMBER_COMPARATORS),
+                },
+                cellRenderer: NumberCellRenderer,
+                ...sortAndFilterProps,
+            }),
+            makeAgGridCustomHeaderColumn({
+                headerName: intl.formatMessage({
+                    id: 'DynamicSimulationTimeLineEventModelName',
+                }),
+                field: COL_MODEL_NAME,
+                width: MEDIUM_COLUMN_WIDTH,
+                filterParams: {
+                    filterDataType: FILTER_DATA_TYPES.TEXT,
+                    filterComparators: [
+                        FILTER_TEXT_COMPARATORS.STARTS_WITH,
+                        FILTER_TEXT_COMPARATORS.CONTAINS,
+                    ],
+                },
+                ...sortAndFilterProps,
+            }),
+            makeAgGridCustomHeaderColumn({
+                headerName: intl.formatMessage({
+                    id: 'DynamicSimulationTimeLineEventModelMessage',
+                }),
+                field: COL_MESSAGE,
+                width: LARGE_COLUMN_WIDTH,
+                filterParams: {
+                    filterDataType: FILTER_DATA_TYPES.TEXT,
+                    filterComparators: [
+                        FILTER_TEXT_COMPARATORS.STARTS_WITH,
+                        FILTER_TEXT_COMPARATORS.CONTAINS,
+                    ],
+                },
+                ...sortAndFilterProps,
+            }),
+        ];
+    }, [intl, sortAndFilterProps]);
+
+    const defaultColDef = useMemo(
+        () => ({
+            filter: true,
+            sortable: true,
+            resizable: true,
+            lockPinned: true,
+            suppressMovable: true,
+            wrapHeaderText: true,
+            autoHeaderHeight: true,
+            cellRenderer: DefaultCellRenderer,
+        }),
+        []
+    );
```

Comment:
You can define it outside of the component

Reply:
DONE!